tests/e2e/regression/test_training_only_acceptance.py

```
@@ -0,0 +1,178 @@
+    """Return a directory containing only the top-level .pt training files.
+
+    ``list_files`` in the training script walks subdirectories, which would
+    pick up non-training files such as ``vocab_mapping/token_freq_sharegpt.pt``.
+    This creates a flat directory of symlinks to avoid that.
+    """
+    data_dir = tmp_path / "training_data"
+    data_dir.mkdir()
+    for item in hidden_states_dir.iterdir():
+        if (
+            item.is_file()
+            and item.suffix == ".pt"
```

Ruff (0.15.10)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)